library/ssl_tls.c

```diff
@@ -1377,14 +1399,36 @@ void ssl_calc_verify_tls_sha256( mbedtls_ssl_context *ssl, unsigned char hash[32
     MBEDTLS_SSL_DEBUG_MSG( 2, ( "<= calc verify" ) );
 
     mbedtls_sha256_free( &sha256 );
-
+#endif /* MBEDTLS_USE_PSA_CRYPTO */
     return;
 }
 #endif /* MBEDTLS_SHA256_C */
 
 #if defined(MBEDTLS_SHA512_C)
 void ssl_calc_verify_tls_sha384( mbedtls_ssl_context *ssl, unsigned char hash[48] )
 {
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    size_t hash_size;
+    psa_status_t status;
+    psa_hash_operation_t sha512_psa = psa_hash_operation_init();
```

Comment:
I'd prefer to call this `sha384_psa`, because it's an artefact of Mbed TLS' MD interface that you use a SHA-512 context for SHA-384 -- at the PSA API boundary, SHA-384 is as different from SHA-512 as any other digest, as far as I see.